sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -568,9 +581,6 @@ object RemoveRedundantAliases extends Rule[LogicalPlan] {
           newChild
         }
 
-        // Create the attribute mapping. Note that the currentNextAttrPairs can contain duplicate
-        // keys in case of Union (this is caused by the PushProjectionThroughUnion rule); in this
-        // case we use the first mapping (which should be provided by the first child).
```

Comment:
since we remove this comment, shall we add an assert to make sure there is no duplicate keys?

Reply:
added in https://github.com/apache/spark/pull/37334/commits/11020eee40af3e1d513b393e33509b13bf6e432c